### DISCUSSION TITLE
Error when setting up db: https://stackoverflow.com/questions/75621212/capacitor-community-sqlite-error-query-failed-selectsql-queryall-no-such-t

### DISCUSSION THREAD

**Discussion by folsze** [capacitor-community/sqlite#384] [Q&A] [answered]
Hi I hope this is the right place to ask this:
https://stackoverflow.com/questions/75621212/capacitor-community-sqlite-error-query-failed-selectsql-queryall-no-such-t

I am a beginner when it comes to working with 3rd party libraries. Maybe someone could give me hints on where to search for the cause of my error?

**Comment by jepiqueau** [comment]
@folsze can you share your code with github

**Reply by folsze** [reply]
I will post a minimal reproducible example here, I will cut away the private stuff from my project

**Reply by folsze** [reply]
almost done. But just a heads up: it's probably a very primitive thing that I forgot.

Oh yeah and: I copied the sql-wasm.wasm simply from the angular starter project, hope that works.

**Comment by jepiqueau** [comment]
@folsze is the error happen in web platform or native platform or both

**Reply by folsze** [reply]
Currently I have only tested web

**Comment by jepiqueau** [comment]
@folsze can you give the link of the minimal reproductible example i will have a look tonight

**Reply by folsze** [reply]
https://github.com/folsze/geo-champ-minified1

thanks a lot! it was hard for me to set this minimal reproducible example up but I now got it done

**Reply by folsze** [reply]
Hey I also created this now:
https://github.com/folsze/ionic-angular-sqlite-starter-minified

on branch main you will find the "ionic-angular-sqlite-starter" without en/decryption and without 
author/posts/products (only with employees-departments)

it was very helpful for me as a beginner because I was overwhelmed with the huge code that was there. I created a very minimal CRUD example. SO: feel free to link this in your projects documentation for beginners! It helped me a lot, cause ~33% of the code disappeared.

on branch without-pages you will find a state of the app that doesn't have any pages at all, it only showcases the database connection (which successfully connects)

**Reply by folsze** [reply]
merci beaucoup d'avance

**Comment by jepiqueau** [comment, marked as answer]
@folsze The error comes from a wrong upgrade-statements it should be:
```
export const mapModeLocationVersionUpgrades = [
  {
    toVersion: 1,
    statements: [
      `CREATE TABLE IF NOT EXISTS map (
          id integer PRIMARY KEY AUTOINCREMENT NOT NULL
        );`,
      `CREATE TABLE IF NOT EXISTS mode (
          id integer PRIMARY KEY AUTOINCREMENT NOT NULL,
          mapId integer,
          FOREIGN KEY (mapId) REFERENCES map (id)
        );`,
    ]
  },
]
```
this make the app working till it reach another bug.
Hope this will help you to start.

**Reply by folsze** [reply]
Thank you so much! This helps me so much, I was looking all over the code for any issues. Surely other following errors hopefully will be resolvable by myself. 

Just curious: is there any way to get some error-message-output on the .addUpgradeStatement method? Because I think it failed if I am not mistaken

**Comment by jepiqueau** [comment]
@folsze i may have a look the way i found the error is to run the ios version where the error message is better handle. Anyhow i wish you à good development. I  do not know why you were installing  sql-wasm as jeep-sqlite use sql.js which came with sql.wasm

**Reply by folsze** [reply]
Oh ok interesting, & very good to know that IOS gives a better error message! I will try soon

I think a better error message would be extremely friendly for beginner users of this library. But I think it's already great, especially the starter projects. First time for me starting with a new big library that only has github projects and no videos on youtube and I still managed to figure out how to use it!

the sql-wasm package was just a relict from me figuring out stuff, I forgot to remove it still
